// lang-js-number/index.js
var e = exports;
var currency = e.currency = {};


currency.af =
 currency.be =
 currency.en_za = num_8cfdfce;

currency.agq =
 currency.kab =
 currency.shi =
 currency.zgh = num_9fbee07;

currency.ak =
 currency.am =
 currency.bem =
 currency.bm =
 currency.cgg =
 currency.chr =
 currency.cy =
 currency.dav =
 currency.ebu =
 currency.ee =
 currency.en =
 currency.es_419 =
 currency.fil =
 currency.ga =
 currency.gd =
 currency.guz =
 currency.gv =
 currency.haw =
 currency.ig =
 currency.ja =
 currency.jmc =
 currency.kam =
 currency.kde =
 currency.ki =
 currency.kln =
 currency.kn =
 currency.ko =
 currency.kw =
 currency.luy =
 currency.mas =
 currency.mer =
 currency.mg =
 currency.ml =
 currency.ms =
 currency.mt =
 currency.naq =
 currency.nd =
 currency.nus =
 currency.nyn =
 currency.om =
 currency.rof =
 currency.saq =
 currency.si =
 currency.sn =
 currency.so =
 currency.sw =
 currency.teo =
 currency.th =
 currency.ti =
 currency.ug =
 currency.vai =
 currency.vun =
 currency.yo =
 currency.zh_hans_hk =
 currency.zh_hans_mo =
 currency.zh_hans_sg =
 currency.zh_hant =
 currency.zh_hant_hk =
 currency.zh_hant_mo =
 currency.zh_hant_tw =
 currency.zu = num_7f26c8c;

currency.ar = num_9a8ef82;

currency.ar_dz =
 currency.ar_ly =
 currency.ar_ma =
 currency.ar_tn =
 currency.az =
 currency.de_at =
 currency.es_py =
 currency.es_uy =
 currency.fur =
 currency.fy =
 currency.jgo =
 currency.kkj =
 currency.mgh =
 currency.mk =
 currency.ms_latn_bn =
 currency.nl =
 currency.nnh =
 currency.qu_bo =
 currency.rw = num_20d37c0;

currency.ar_eh =
 currency.bo =
 currency.ha =
 currency.ii =
 currency.lag =
 currency.lkt =
 currency.mgo =
 currency.mn =
 currency.qu =
 currency.root =
 currency.sah =
 currency.smn =
 currency.ta_my =
 currency.ta_sg =
 currency.to =
 currency.yi =
 currency.zh = num_2c50f52;

currency.ar_lb =
 currency.ar_mr = num_78f5b74;

currency.as = num_6a9eee0;

currency.asa =
 currency.he =
 currency.xog = num_276e645;

currency.ast =
 currency.bs =
 currency.ca =
 currency.da =
 currency.de =
 currency.dsb =
 currency.el =
 currency.en_150 =
 currency.en_be =
 currency.es =
 currency.eu =
 currency.fr_be =
 currency.fr_lu =
 currency.hr =
 currency.hsb =
 currency.is =
 currency.it =
 currency.lb =
 currency.ln =
 currency.nl_be =
 currency.ro =
 currency.sl =
 currency.sr =
 currency.tr =
 currency.vi = num_2ff75d7;

currency.bas =
 currency.bg =
 currency.br =
 currency.cs =
 currency.dua =
 currency.dyo =
 currency.et =
 currency.ewo =
 currency.ff =
 currency.fi =
 currency.fr =
 currency.hu =
 currency.ka =
 currency.kea =
 currency.kk =
 currency.ksf =
 currency.ksh =
 currency.ky =
 currency.lt =
 currency.nmg =
 currency.nn =
 currency.pl =
 currency.pt_pt =
 currency.ru =
 currency.se =
 currency.sk =
 currency.sq =
 currency.sv =
 currency.tzm =
 currency.uk =
 currency.yav = num_19ba5e1;

currency.bez =
 currency.ksb =
 currency.lg =
 currency.luo =
 currency.rwk =
 currency.sbp = num_716d133;

currency.bn = num_44fbe6a;

currency.brx =
 currency.en_in =
 currency.kok =
 currency.or =
 currency.pa =
 currency.ta =
 currency.ur = num_1acbcfc;

currency.de_ch =
 currency.de_li =
 currency.it_ch = num_3911370;

currency.dje =
 currency.khq =
 currency.ses =
 currency.twq = num_fa6ce51;

currency.dz = num_a0f41f0;

currency.en_us_posix = num_7adcc8e;

currency.eo =
 currency.nb =
 currency.os =
 currency.uz = num_8d79dc0;

currency.es_cl =
 currency.es_ec =
 currency.es_gq =
 currency.es_ve =
 currency.fo =
 currency.gl =
 currency.id =
 currency.kl =
 currency.km =
 currency.lo =
 currency.mua =
 currency.pt =
 currency.sg =
 currency.sw_cd = num_9ca9b41;

currency.fa = num_b97b5d8;

currency.fr_ch =
 currency.mfe = num_d914f88;

currency.gsw =
 currency.rm = num_cf4f10d;

currency.gu =
 currency.hi =
 currency.te = num_298905f;

currency.hy = num_e6d7ef6;

currency.ks =
 currency.pa_arab =
 currency.pa_arab_pk =
 currency.ur_in = num_e8a8265;

currency.lu =
 currency.rn =
 currency.seh = num_48f4f05;

currency.lv = num_e7509f5;

currency.mr = num_b3dbad2;

currency.my = num_dc41966;

currency.ne = num_bfadaad;

currency.ps =
 currency.uz_arab =
 currency.uz_arab_af = num_76189e4;

currency.wae = num_5d6b734;
var decimal = e.decimal = {};


decimal.af =
 decimal.agq =
 decimal.bas =
 decimal.be =
 decimal.bg =
 decimal.br =
 decimal.cs =
 decimal.dua =
 decimal.dyo =
 decimal.en_za =
 decimal.eo =
 decimal.et =
 decimal.ewo =
 decimal.ff =
 decimal.fi =
 decimal.fr =
 decimal.hu =
 decimal.ka =
 decimal.kab =
 decimal.kea =
 decimal.kk =
 decimal.ksf =
 decimal.ksh =
 decimal.ky =
 decimal.lt =
 decimal.lv =
 decimal.nb =
 decimal.nmg =
 decimal.nn =
 decimal.os =
 decimal.pl =
 decimal.pt_pt =
 decimal.ru =
 decimal.se =
 decimal.shi =
 decimal.sk =
 decimal.sq =
 decimal.sv =
 decimal.tzm =
 decimal.uk =
 decimal.uz =
 decimal.yav =
 decimal.zgh = num_33b8a03;

decimal.ak =
 decimal.am =
 decimal.ar_eh =
 decimal.asa =
 decimal.bem =
 decimal.bez =
 decimal.bm =
 decimal.bo =
 decimal.cgg =
 decimal.chr =
 decimal.cy =
 decimal.dav =
 decimal.ebu =
 decimal.ee =
 decimal.en =
 decimal.es_419 =
 decimal.fil =
 decimal.ga =
 decimal.gd =
 decimal.guz =
 decimal.gv =
 decimal.ha =
 decimal.haw =
 decimal.he =
 decimal.ig =
 decimal.ii =
 decimal.ja =
 decimal.jmc =
 decimal.kam =
 decimal.kde =
 decimal.ki =
 decimal.kln =
 decimal.kn =
 decimal.ko =
 decimal.ksb =
 decimal.kw =
 decimal.lag =
 decimal.lg =
 decimal.lkt =
 decimal.luo =
 decimal.luy =
 decimal.mas =
 decimal.mer =
 decimal.mg =
 decimal.mgo =
 decimal.mn =
 decimal.ms =
 decimal.mt =
 decimal.naq =
 decimal.nd =
 decimal.nus =
 decimal.nyn =
 decimal.om =
 decimal.qu =
 decimal.rof =
 decimal.root =
 decimal.rwk =
 decimal.sah =
 decimal.saq =
 decimal.sbp =
 decimal.si =
 decimal.smn =
 decimal.sn =
 decimal.so =
 decimal.sw =
 decimal.ta_my =
 decimal.ta_sg =
 decimal.teo =
 decimal.th =
 decimal.ti =
 decimal.to =
 decimal.ug =
 decimal.ur =
 decimal.vai =
 decimal.vun =
 decimal.xog =
 decimal.yi =
 decimal.yo =
 decimal.zh =
 decimal.zu = num_106f234;

decimal.ar = num_32fde70;

decimal.ar_dz =
 decimal.ar_ly =
 decimal.ar_ma =
 decimal.ar_tn =
 decimal.ast =
 decimal.az =
 decimal.bs =
 decimal.ca =
 decimal.da =
 decimal.de =
 decimal.dsb =
 decimal.el =
 decimal.en_150 =
 decimal.en_be =
 decimal.es =
 decimal.eu =
 decimal.fo =
 decimal.fr_be =
 decimal.fr_lu =
 decimal.fur =
 decimal.fy =
 decimal.gl =
 decimal.hr =
 decimal.hsb =
 decimal.id =
 decimal.is =
 decimal.it =
 decimal.jgo =
 decimal.kkj =
 decimal.kl =
 decimal.km =
 decimal.lb =
 decimal.ln =
 decimal.lo =
 decimal.lu =
 decimal.mgh =
 decimal.mk =
 decimal.ms_latn_bn =
 decimal.mua =
 decimal.nl =
 decimal.nnh =
 decimal.pt =
 decimal.qu_bo =
 decimal.rn =
 decimal.ro =
 decimal.rw =
 decimal.seh =
 decimal.sg =
 decimal.sl =
 decimal.sr =
 decimal.sw_cd =
 decimal.tr =
 decimal.vi = num_0ea8d48;

decimal.ar_lb =
 decimal.ar_mr = num_fe59da6;

decimal.as =
 decimal.bn = num_9cd76d7;

decimal.brx =
 decimal.en_in =
 decimal.gu =
 decimal.hi =
 decimal.kok =
 decimal.ml =
 decimal.or =
 decimal.pa =
 decimal.ta =
 decimal.te = num_07fc74c;

decimal.de_ch =
 decimal.de_li =
 decimal.it_ch = num_e39b9c5;

decimal.dje =
 decimal.fr_ch =
 decimal.khq =
 decimal.mfe =
 decimal.ses =
 decimal.twq = num_77abe6b;

decimal.dz = num_b87db83;

decimal.en_us_posix = num_56bbc40;

decimal.fa =
 decimal.ur_in = num_ef15e72;

decimal.gsw =
 decimal.rm = num_91e7892;

decimal.hy = num_4866f6b;

decimal.ks =
 decimal.pa_arab =
 decimal.pa_arab_pk = num_72a5f9c;

decimal.mr = num_1f391bf;

decimal.my = num_c5b22c5;

decimal.ne = num_87c9b5a;

decimal.ps =
 decimal.uz_arab =
 decimal.uz_arab_af = num_af4bea9;

decimal.wae = num_a04bb59;
var percent = e.percent = {};


percent.af =
 percent.agq =
 percent.be =
 percent.bg =
 percent.dyo =
 percent.en_za =
 percent.eo =
 percent.et =
 percent.ewo =
 percent.ff =
 percent.hu =
 percent.kab =
 percent.kea =
 percent.kk =
 percent.ksf =
 percent.ky =
 percent.lv =
 percent.nmg =
 percent.os =
 percent.pl =
 percent.pt_pt =
 percent.shi =
 percent.sq =
 percent.tzm =
 percent.uk =
 percent.uz =
 percent.yav = num_1bf3c2c;

percent.ak =
 percent.am =
 percent.ar_eh =
 percent.asa =
 percent.bem =
 percent.bez =
 percent.bm =
 percent.bo =
 percent.cgg =
 percent.chr =
 percent.cy =
 percent.dav =
 percent.ebu =
 percent.ee =
 percent.en =
 percent.fil =
 percent.ga =
 percent.gd =
 percent.guz =
 percent.gv =
 percent.ha =
 percent.haw =
 percent.he =
 percent.ig =
 percent.ii =
 percent.ja =
 percent.jmc =
 percent.kam =
 percent.kde =
 percent.ki =
 percent.kln =
 percent.kn =
 percent.ko =
 percent.ksb =
 percent.kw =
 percent.lag =
 percent.lg =
 percent.lkt =
 percent.luo =
 percent.luy =
 percent.mas =
 percent.mer =
 percent.mg =
 percent.mgo =
 percent.ml =
 percent.mn =
 percent.ms =
 percent.mt =
 percent.naq =
 percent.nd =
 percent.nus =
 percent.nyn =
 percent.om =
 percent.rof =
 percent.root =
 percent.rwk =
 percent.sah =
 percent.saq =
 percent.sbp =
 percent.si =
 percent.smn =
 percent.sn =
 percent.so =
 percent.sw =
 percent.ta_my =
 percent.ta_sg =
 percent.te =
 percent.teo =
 percent.th =
 percent.ti =
 percent.to =
 percent.ug =
 percent.vai =
 percent.vun =
 percent.xog =
 percent.yi =
 percent.yo =
 percent.zh =
 percent.zu = num_e19905a;

percent.ar = num_dde41b1;

percent.ar_dz =
 percent.ar_ly =
 percent.ar_ma =
 percent.ar_tn =
 percent.ast =
 percent.az =
 percent.bs =
 percent.ca =
 percent.el =
 percent.en_150 =
 percent.en_be =
 percent.es_mx =
 percent.fur =
 percent.fy =
 percent.gl =
 percent.hr =
 percent.id =
 percent.is =
 percent.it =
 percent.jgo =
 percent.kkj =
 percent.km =
 percent.ln =
 percent.lo =
 percent.lu =
 percent.mgh =
 percent.mk =
 percent.ms_latn_bn =
 percent.mua =
 percent.nl =
 percent.nnh =
 percent.pt =
 percent.rw =
 percent.seh =
 percent.sg =
 percent.sl =
 percent.sr =
 percent.sw_cd =
 percent.vi = num_00cb85c;

percent.ar_lb =
 percent.ar_mr = num_4748131;

percent.as =
 percent.bn = num_dfc6a48;

percent.bas =
 percent.br =
 percent.cs =
 percent.dua =
 percent.fi =
 percent.fr =
 percent.ka =
 percent.ksh =
 percent.lt =
 percent.nb =
 percent.nn =
 percent.ru =
 percent.se =
 percent.sk =
 percent.sv =
 percent.zgh = num_f2d2f76;

percent.brx =
 percent.en_in =
 percent.gu =
 percent.hi =
 percent.kok =
 percent.or =
 percent.pa =
 percent.ta =
 percent.ur = num_943e081;

percent.da =
 percent.de =
 percent.dsb =
 percent.es =
 percent.fo =
 percent.fr_be =
 percent.fr_lu =
 percent.hsb =
 percent.kl =
 percent.lb =
 percent.qu_bo =
 percent.rn =
 percent.ro = num_f293813;

percent.de_ch =
 percent.de_li = num_672cf69;

percent.dje =
 percent.khq =
 percent.mfe =
 percent.ses =
 percent.twq = num_9c917c9;

percent.dz = num_9f084f5;

percent.en_us_posix = num_fd94fa9;

percent.es_419 =
 percent.qu = num_b6fede1;

percent.eu = num_a7bb182;

percent.fa = num_7e96de0;

percent.fr_ch = num_6bfcdf8;

percent.gsw =
 percent.rm = num_e54821e;

percent.hy = num_46f5582;

percent.it_ch = num_4e872f4;

percent.ks =
 percent.pa_arab =
 percent.pa_arab_pk =
 percent.ur_in = num_f8da03f;

percent.mr =
 percent.ne = num_3b990c0;

percent.my = num_978fb85;

percent.ps =
 percent.uz_arab =
 percent.uz_arab_af = num_a2f8101;

percent.tr = num_ea61450;

percent.wae = num_bfa5185;
function num_3911370(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "'") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_4748131(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_arab(decimal ? formattedWhole + "," + "" : formattedWhole);
}

function num_8cfdfce(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_33b8a03(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "," + "" + decimal + "" : formattedWhole;
}

function num_1bf3c2c(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

function num_9fbee07(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "" + (opts.unit || "") + "";
}

function num_7f26c8c(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_106f234(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole;
}

function num_e19905a(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

var system_arab_digits = ["٠","١","٢","٣","٤","٥","٦","٧","٨","٩"];
function system_arab(str) {
  return str.replace(/(\d)/g, function(i) { return system_arab_digits[i]; });
};


function num_9a8ef82(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_arab("" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_32fde70(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_arab(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_dde41b1(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_arab(decimal ? formattedWhole + "." + "" : formattedWhole);
}

function num_20d37c0(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_0ea8d48(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "," + "" + decimal + "" : formattedWhole;
}

function num_00cb85c(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

function num_2c50f52(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_78f5b74(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  return system_arab("" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_fe59da6(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_arab(decimal ? formattedWhole + "," + "" + decimal + "" : formattedWhole);
}

var system_beng_digits = ["০","১","২","৩","৪","৫","৬","৭","৮","৯"];
function system_beng(str) {
  return str.replace(/(\d)/g, function(i) { return system_beng_digits[i]; });
};


function num_6a9eee0(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_beng("" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_9cd76d7(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_beng(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_dfc6a48(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_beng(decimal ? formattedWhole + "." + "" : formattedWhole);
}

function num_276e645(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + " " + (opts.unit || "") + "";
}

function num_2ff75d7(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + " " + (opts.unit || "") + "";
}

function num_19ba5e1(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + " " + (opts.unit || "") + "";
}

function num_f2d2f76(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

function num_716d133(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "" + (opts.unit || "") + "";
}

function num_44fbe6a(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_beng("" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "" + (opts.unit || "") + "");
}

function num_1acbcfc(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_07fc74c(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole;
}

function num_943e081(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_f293813(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

function num_e39b9c5(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "'") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole;
}

function num_672cf69(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "'") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_fa6ce51(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "" + (opts.unit || "") + "";
}

function num_77abe6b(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole;
}

function num_9c917c9(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

var system_tibt_digits = ["༠","༡","༢","༣","༤","༥","༦","༧","༨","༩"];
function system_tibt(str) {
  return str.replace(/(\d)/g, function(i) { return system_tibt_digits[i]; });
};


function num_a0f41f0(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_tibt("" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_b87db83(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_tibt(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_9f084f5(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return system_tibt(decimal ? formattedWhole + "." + "" : formattedWhole);
}

function num_7adcc8e(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_56bbc40(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole;
}

function num_fd94fa9(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_8d79dc0(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_b6fede1(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_9ca9b41(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_a7bb182(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + "%" + " " + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

var system_arabext_digits = ["۰","۱","۲","۳","۴","۵","۶","۷","۸","۹"];
function system_arabext(str) {
  return str.replace(/(\d)/g, function(i) { return system_arabext_digits[i]; });
};


function num_b97b5d8(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_arabext("‎" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_ef15e72(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_arabext(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_7e96de0(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_arabext(decimal ? formattedWhole + "." + "" : formattedWhole);
}

function num_d914f88(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_6bfcdf8(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + " ") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_cf4f10d(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "’") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + " " + (opts.unit || "") + "";
}

function num_91e7892(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "’") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole;
}

function num_e54821e(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "’") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + " " + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_298905f(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_e6d7ef6(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  return "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + " " + (opts.unit || "") + "";
}

function num_4866f6b(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "," + "" + decimal + "" : formattedWhole;
}

function num_46f5582(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

function num_4e872f4(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "'") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "." + "" : formattedWhole;
}

function num_e8a8265(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_arabext("" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_72a5f9c(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_arabext(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_f8da03f(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_arabext(decimal ? formattedWhole + "." + "" : formattedWhole);
}

function num_48f4f05(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  return "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "" + (opts.unit || "") + "";
}

function num_e7509f5(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  return "" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

var system_deva_digits = ["०","१","२","३","४","५","६","७","८","९"];
function system_deva(str) {
  return str.replace(/(\d)/g, function(i) { return system_deva_digits[i]; });
};


function num_b3dbad2(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_deva("" + (opts.unit || "") + "" + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_1f391bf(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 2 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_deva(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_3b990c0(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_deva(decimal ? formattedWhole + "." + "" : formattedWhole);
}

var system_mymr_digits = ["၀","၁","၂","၃","၄","၅","၆","၇","၈","၉"];
function system_mymr(str) {
  return str.replace(/(\d)/g, function(i) { return system_mymr_digits[i]; });
};


function num_dc41966(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_mymr("" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_c5b22c5(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_mymr(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_978fb85(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_mymr(decimal ? formattedWhole + "." + "" : formattedWhole);
}

function num_bfadaad(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  return system_deva("" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +"."+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "");
}

function num_87c9b5a(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ",") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_deva(decimal ? formattedWhole + "." + "" + decimal + "" : formattedWhole);
}

function num_76189e4(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  return system_arabext("" + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + " " + (opts.unit || "") + "");
}

function num_af4bea9(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return system_arabext(decimal ? formattedWhole + "," + "" + decimal + "" : formattedWhole);
}

function num_a2f8101(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return system_arabext(decimal ? formattedWhole + "," + "" : formattedWhole);
}

function num_ea61450(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + ".") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + "%" + "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}

function num_5d6b734(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1] || '';
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "’") + newWhole;
  }
  whole = newWhole;
  return "" + (opts.unit || "") + " " + (!whole.length ? "0" : whole) + "" +","+"" + (decimal.length < 2 ? (decimal + "00").slice(0, 2) : decimal) + "";
}

function num_a04bb59(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "’") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "";
  return decimal ? formattedWhole + "," + "" + decimal + "" : formattedWhole;
}

function num_bfa5185(amount, opts) {
  opts = opts || {};
  var parts = (amount + '').split('.');
  var whole = parts[0];
  var decimal = parts[1];
  for (var i=0, newWhole="", l = whole.length - 1; i <= l; i++) {
    newWhole = (!i || i % 3 ? whole[l - i] : whole[l - i] + "’") + newWhole;
  }
  whole = newWhole;
  var formattedWhole = "" + (!whole.length ? "0" : whole) + "" + "%" + "";
  return decimal ? formattedWhole + "," + "" : formattedWhole;
}
